src/renderer/services/binance/ledger.ts

```diff
@@ -1,27 +1,86 @@
 import * as RD from '@devexperts/remote-data-ts'
+import { Client } from '@xchainjs/xchain-binance'
+import { Address, TxParams } from '@xchainjs/xchain-client'
+import * as E from 'fp-ts/Either'
 import * as FP from 'fp-ts/lib/function'
+import * as O from 'fp-ts/Option'
 import * as Rx from 'rxjs'
-import { catchError, map, startWith } from 'rxjs/operators'
+import { catchError, map, startWith, switchMap } from 'rxjs/operators'
 
-import { Network } from '../../../shared/api/types'
+import { LedgerBNBTxInfo, Network } from '../../../shared/api/types'
 import { observableState } from '../../helpers/stateHelper'
-import { LedgerAddressRD } from '../wallet/types'
+import { XChainClient$ } from '../clients'
+import { ErrorId, LedgerAddressRD, LedgerTxLD, LedgerTxRD } from '../wallet/types'
 import { LedgerService } from './types'
 
-const { get$: ledgerAddress$, set: setLedgerAddressRD } = observableState<LedgerAddressRD>(RD.initial)
+export const createLedgerService = <T extends XChainClient$<Client>>(client$: T): LedgerService => {
+  const { get$: ledgerAddress$, set: setLedgerAddressRD } = observableState<LedgerAddressRD>(RD.initial)
 
-const retrieveLedgerAddress = (network: Network) =>
-  FP.pipe(
-    Rx.from(window.apiHDWallet.getLedgerAddress('BNB', network)),
-    map(RD.fromEither),
-    startWith(RD.pending),
-    catchError((error) => Rx.of(RD.failure(error)))
-  ).subscribe(setLedgerAddressRD)
+  const retrieveLedgerAddress = (network: Network) =>
+    FP.pipe(
+      Rx.from(window.apiHDWallet.getLedgerAddress('BNB', network)),
+      map(RD.fromEither),
+      startWith(RD.pending),
+      catchError((error) => Rx.of(RD.failure(error)))
+    ).subscribe(setLedgerAddressRD)
```

Comment:
[nitpick] It looks very similar to BTC implementation of `retrieveLedgerAddress`. We can provide a more general function such as 

```
const retrieveLedgerAddressByChain = (chain: Chain, network: Network)) ...
```
to use in `binance/ledger` and in `bitcoin/ledger`. _But that's not really needed for now ..._